docs/architecture.rst

```diff
@@ -0,0 +1,189 @@
+ How ppb Fits Together
+===========================================================
+
+This document is an overall reference of what ppb is, how it's organized, and
+how to use the various pieces.
+
+The Conceptual Necessities
```

Comment:
This section is really well written -- I love it!
Only suggestion I have is to perhaps put a list of all the major topics to be covered right up front, something like:
- GameObject
- GameEngine
- Systems
- Scenes
- Events
- Assets